simperby/src/lib.rs

```diff
@@ -98,30 +176,18 @@ pub trait SimperbyApi {
     async fn get_height(&self) -> u64;
 
     /// Gets the finalized block for the given height.
-    async fn get_block(&self, height: u64) -> Result<Block, String>;
+    async fn get_block(&self, height: u64) -> Result<Block, SimperbyError>;
 
-    /// Checks the given block as the next block to be added to the current state.
+    /// Checks the given block as the next block to be added to the block of the given `height`.
     ///
     /// Fails if the block is invalid.
-    async fn check_block(&self, block: Block) -> Result<(), String>;
-
-    /// Attempts to propose a block for this round.
-    ///
-    /// It fails
-    /// 1. with the same cause as `check_block`
-    /// 2. if this node is not the current leader.
-    /// 3. if this node has already proposed another block for this round.
-    async fn propose_block(
-        &self,
-        block: Block,
-        signature: TypedSignature<Block>,
-    ) -> Result<(), String>;
+    async fn check_block(&self, block: Block, height: BlockHeight) -> Result<(), SimperbyError>;
 
     /// Reads the finlized state entry by the given key.
-    async fn read_state(&self, key: String, height: u64) -> Result<Vec<u8>, String>;
+    async fn read_state(&self, key: String, height: BlockHeight) -> Result<Vec<u8>, SimperbyError>;
 
     /// Gets the current possible consensus voting options.
-    async fn get_consensus_vote_options(&self) -> Vec<ConsensusVoteItem>;
+    async fn get_consensus_vote_options(&self) -> Result<Vec<ConsensusVoteItem>, SimperbyError>;
```

Comment:
```suggestion
    async fn get_consensus_voting_options(&self) -> Result<Vec<ConsensusVoteItem>, SimperbyError>;
```

Reply:
We have used `vote` in many places. Let's just keep it.